libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -67,6 +67,30 @@ assembly::Statement Parser::parseStatement()
 		return parseFunctionDefinition();
 	case Token::LBrace:
 		return parseBlock();
+	case Token::Switch:
+	{
+		if (!m_julia)
+			break;
+		assembly::Switch switchcase = createWithLocation<assembly::Switch>();
+		m_scanner->next();
+		assembly::Statement statement = parseExpression();
+		if (statement.type() == typeid(assembly::FunctionCall))
+			switchcase.expression = boost::get<assembly::FunctionCall>(statement);
+		if (statement.type() == typeid(assembly::Identifier))
+			switchcase.expression = boost::get<assembly::Identifier>(statement);
+		if (statement.type() == typeid(assembly::Literal))
+			switchcase.expression = boost::get<assembly::Literal>(statement);
+//		switchcase.expression = parseExpression();
+		while (m_scanner->currentToken() == Token::Case)
+			switchcase.cases.emplace_back(parseCase());
+		if (m_scanner->currentToken() == Token::Default)
```

Comment:
We agreed to that and the grammar reflects this restriction. I think it makes sense, given it is the "fallback".